projects/vdk-plugins/vdk-kerberos-auth/src/vdk/plugin/kerberos/kerberos_plugin.py

```diff
@@ -0,0 +1,121 @@
+# Copyright 2021 VMware, Inc.
+# SPDX-License-Identifier: Apache-2.0
+import logging
+import os
+
+from vdk.api.plugin.hook_markers import hookimpl
+from vdk.internal.builtin_plugins.run.job_context import JobContext
+from vdk.internal.control.exception.vdk_exception import VDKException
+from vdk.internal.core import errors
+from vdk.internal.core.config import ConfigurationBuilder
+from vdk.plugin.kerberos.kinit_authenticator import KinitGSSAPIAuthenticator
+from vdk.plugin.kerberos.minikerberos_authenticator import (
+    MinikerberosGSSAPIAuthenticator,
+)
+
+
+KRB_AUTH = "KRB_AUTH"
+KEYTAB_FOLDER = "KEYTAB_FOLDER"
+KEYTAB_FILENAME = "KEYTAB_FILENAME"
+KEYTAB_PRINCIPAL = "KEYTAB_PRINCIPAL"
+KEYTAB_REALM = "KEYTAB_REALM"
+KERBEROS_KDC_HOST = "KERBEROS_KDC_HOST"
+
+
+class KerberosPlugin:
```

Comment:
Let's name it KerberosAuthenticatorFactory or something like this. 

XXXPlugin are most often classes with hook implementations inside.

Reply:
I refactored the code a little bit taking this suggestion into consideration. The authenticators now share a common base class and are instantiated through a factory based on the authentication type.